inference_segmentation.py

```diff
@@ -225,26 +227,22 @@ def segmentation(param,
     del fp
 
     fp = np.memmap(tp_mem, dtype='float16', mode='r', shape=(tf_len, h_padded, w_padded, num_classes))
-    pred_img = np.zeros((h_padded, w_padded), dtype=np.uint8)
+    pred_heatmap = np.zeros((h_padded, w_padded, num_classes), dtype=np.float16)
     for row, col in tqdm(itertools.product(range(0, input_image.height, chunk_size),
                                            range(0, input_image.width, chunk_size)),
                          leave=False, total=total_inf_windows, desc="Writing to array"):
         arr1 = (fp[:, row:row + chunk_size, col:col + chunk_size, :]).max(axis=0)
         if single_class_mode:
             arr1 = sigmoid(arr1)
-            arr1 = (arr1 > threshold)
-            arr1 = np.squeeze(arr1, axis=2).astype(np.uint8)
-        else:
-            arr1 = np.argmax(arr1, axis=-1).astype(np.uint8)
-        pred_img[row:row + chunk_size, col:col + chunk_size] = arr1
+        pred_heatmap[row:row + chunk_size, col:col + chunk_size, :] = arr1
 
     end_seg = time.time() - start_seg
     logging.info('Segmentation operation completed in {:.0f}m {:.0f}s'.format(end_seg // 60, end_seg % 60))
 
     if debug:
-        logging.debug(f'Bin count of final output: {np.unique(pred_img, return_counts=True)}')
+        logging.debug(f'Bin count of final output: {np.unique(pred_heatmap, return_counts=True)}')
```

Comment:
This will need to be adjusted as np.unique will go crazy on the amount of values contained in the pred_heatmap since this array is float16, not uint8.

Reply:
This is ok since datatype will not be unsigned integer.